### DISCUSSION TITLE
When the load model size is too large, result in the app crashes.

### DISCUSSION THREAD

**Discussion by DavidWeng916** [pixiv/vroid-sdk-developers#88] [Q&A] [answered]
Hello, I would like to ask a question, I recently found some problems when I load the model, if the size of the model is too large, It seems that Unity's ManagedHeap will grow very fast and cause the app to crash as a result.

I added these two models to Hearts to test
[Murder Drones / N](https://hub.vroid.com/en/characters/8867639520850271942/models/6764553087108701082)
[Murder Drones / Uzi Doorman](https://hub.vroid.com/en/characters/8867639520850271942/models/6524463618192331076)

This is the state when no model has been read yet
<img width="1327" alt="截圖 2023-06-13 下午5 43 35" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/67bf7a76-cf3e-4a3c-818f-7f9fb0f86a1c">

This is the state after loading the above two models
<img width="1326" alt="截圖 2023-06-13 下午5 51 47" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/0857bf53-b6bf-4181-b12c-82e023d7af7f">

When I kept switching between the two models, I could see through the Unity Profiler that the Managed Heap would keep going up and up, and eventually, the app would crash.

I'm using VRoid SDK 0.2.0 with the example project (CharacterModelExample) in iPhone XR can be easily reproduced, in addition I test other apps using VRoid SDK and also crash, I would like to ask if there is any solution to this problem?

**Comment by nyamadan** [comment, marked as answer]
I have no idea how to load large VRM models with small heap.

If you check the [original_file_size](https://developer.vroid.com/en/sdk/docs/0.2.2/Pixiv.VroidSdk.Api.DataModel.CharacterModelVersion.html#Pixiv_VroidSdk_Api_DataModel_CharacterModelVersion_original_file_size), you can prevent large VRM models from being read.

**Reply by DavidWeng916** [reply]
> I have no idea how to load large VRM models with small heap.

Hello, as I mentioned above, I added these two models to the Love Heart.
[Murder Drones / N](https://hub.vroid.com/en/characters/8867639520850271942/models/6764553087108701082)
[Murder Drones / Uzi Doorman](https://hub.vroid.com/en/characters/8867639520850271942/models/6524463618192331076)
<img width="1612" alt="截圖 2023-06-15 上午11 26 04" src="https://github.com/pixiv/vroid-sdk-developers/assets/115971193/8b0cbc26-f363-40b4-b59e-d8a8b7e55156">

So I can use the DefaultApi.GetHeartsAsync() API to get this model and load this model, and I can reproduce the crash issue using iPhone XR.

> If you check the [original_file_size](https://developer.vroid.com/en/sdk/docs/0.2.2/Pixiv.VroidSdk.Api.DataModel.CharacterModelVersion.html#Pixiv_VroidSdk_Api_DataModel_CharacterModelVersion_original_file_size), you can prevent large VRM models from being read.

Okay, we can try this approach, thank you.
